lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -150,13 +154,28 @@ static bool validateCodingKeysEnum(DerivedConformance &derived,
   for (auto *varDecl : derived.Nominal->getStoredProperties()) {
     if (!varDecl->isUserAccessible())
       continue;
-
-    properties[getVarNameForCoding(varDecl)] = varDecl;
+    // Redeclarations(which are invalid) are not relevant to coding
+    // keys validation.
+    const auto varIdentifier = getVarNameForCoding(varDecl);
+    if (properties.count(varIdentifier))
+      continue;
+      
+    properties[varIdentifier] = varDecl;
   }
-
+  
+  llvm::DenseSet<Identifier> validatedProperties;
```

Comment:
Sorry I didn't mention this earlier – could you use a `llvm::SmallDenseSet<Identifier, 8>` to match the map expectations?